core/plug-in/codec2/codec2.c

```diff
@@ -0,0 +1,298 @@
+#include "codec2.h"
+#include "amci.h"
+#include "codecs.h"
+#include "../../log.h"
+#include <codec2/codec2.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+
+/**
+ * In order to tell SEMS to buffer the data and invoke the pcm16_2_xyz function
+ * only after 20ms or 40ms we should specify parameters (frame length, frame size
+ * and encoded frame size) here.
+ * In order to figure out what should be AMCI_FMT_ENCODED_FRAME_SIZE see Codec2's
+ * codec2_bits_per_frame function in ${CODEC2_ROOT}/src/codec2.c file.
+ *
+ * It should be calculated like this (assuming that codec2 was already created):
+ *
+ * const int bits_per_frame = codec2_bits_per_frame(CODEC2);
+ * const int AMCI_FMT_ENCODED_FRAME_SIZE = (bits_per_frame + 7) / 8;
+ */
+static amci_codec_fmt_info_t codec2_fmt_description_20ms_3200[] = { {AMCI_FMT_FRAME_LENGTH, 20},
+                                                                    {AMCI_FMT_FRAME_SIZE, 160},
+                                                                    {AMCI_FMT_ENCODED_FRAME_SIZE, 8}, {0,0}};
```

Comment:
I would still like to find a way to obtain the number of bytes per payload from libcodec2. Although there is nothing wrong with hardcoding those numbers here, I am a little concerned about the numbers getting out of sync if libcodec2 is updated. In my option, there should be a single source of truth for parameters like these.

Reply:
I am working on it.

Reply:
Fixed, see https://github.com/columbia-irt/sems/blob/irt_lab_add_codec2/core/plug-in/codec2/codec2.c#L114-L123